### ISSUE TITLE
Incorrect cursore pointer for "More" option in navigation bar

### ISSUE THREAD

**Issue by rohanharikr** [mozilla/addons-frontend#10549]
### Describe the problem and steps to reproduce it:

Hover your cursor over the "More..." option in the navigation bar.

### What happened?

Cursor pointer does not change.

### What did you expect to happen?

Cursor pointer to be type `pointer`.

### Anything else we should know?

The other options in the navigation are type links and have the `cursor: pointer` property by default. Adding the same property for the `.DropdownMenu-button` class / "More..." button should fix the issue.

<img width="299" alt="Screenshot 2021-05-07 at 2 16 19 PM" src="https://user-images.githubusercontent.com/12775813/117424129-17add200-af3f-11eb-8e81-97409931ee90.png">

**Comment by championshuttler** [comment]
good catch @rohanharikr thanks, adding `cursor: pointer` is solving the issue indeed, WDYT @bobsilverberg

**Comment by ioanarusiczki** [comment]
@willdurand Verified on AMO dev - FF88(win10). 

![to pointer](https://user-images.githubusercontent.com/33448286/117943750-520dcb00-b2c1-11eb-91f6-b90b484de850.gif)
